---
title: "final_project_markdown"
author: "Eric Lyall"
date: '2022-03-27'
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Downloading files from SRA

First, we have to download the sequences from SRA. 

```{r}
# using conda to install the sra tools package
#conda instal -c bioconda sra-tools

# fastq-dump SRR10323947 --split-files gets us the last sample
```

## Downloading the reference genome:

We need to download the reference genome to use as a comparison while searching for mutations. The reference genome can be downloaded from NCBI here: https://www.ncbi.nlm.nih.gov/nuccore/NC_000866
Following the instructions from the breseq website, we first need to make sure the "show sequence" box is selected, so that we download the sequence as well as the features:
![Accessing reference genome on GeneBank](GenBank.png)
At this point, we can click the "send to" button and download the file as a genebank file (.gb)
![Accessing reference genome on GeneBank](GenBankDownload.png)
Using scp, we can trasfer the file over to our cluster: 
```{r pressure, echo=FALSE}
#scp sequence.gb elyall_bmeg22@orca1.bcgsc.ca:/home/elyall_bmeg22/final_project/paired
```

After transferring over the file, we need to check to make sure it contains the sequence as well as the features. This can be done by opening the file, and scrolling until you see a header called "Origin". Under this should be a sequence with nucleotides. 


```{r}
#cat sequence.gb

```
![Accessing reference genome on GeneBank](view_genebank_file.png)

#Doing quality control, trimmomatic on fastq files: 

First we can use fastQC to check the quality of our files: 
```{r}
#fastqc SRR10323947_1.fastq

```

The read quality drops off significantly near the ends of reads. This means we will have to use trimmomatic to remove some of the poor quality bases near the end of every read. This allows downstream ananlysis with breseq to only work with high quality data. 
![Base quality has significant drop off near the ends of reads](fastqc_example.png)

Next we can use trimmomatic to trim off the adapter sequences and poor reads near the end. The paper mentions: "Demultiplexed reads were trimmed for Nextera adapter sequences using Trimmomatic with default settings". We also have to trim off the illumina adapters from our reads, as shows in the photo below
![Adapters to get rid of](fastqc_adapters_to_trim.png)

```{r}
#conda install -c bioconda trimmomatic

#First we have to make sure the fastqc files we have are in the zipped format- this makes it faster for trimmomatic. 
#gzip *.fastq

#NExt we need to copy the illumina adapters for nextera paired-end sequencing into our working directory. These are downloaded with trimmomatic. 
#(bmeg400e_env) elyall_bmeg22@orca01:~/anaconda3/pkgs/trimmomatic-0.39-hdfd78af_2/share/trimmomatic/adapters$ cp NexteraPE-PE.fa /home/elyall_bmeg22/final_project/

#Now we can run the trimmomatic commands to remove the illumina adapters. 

#trimmomatic PE SRR10323947_1.fastq.gz SRR10323947_2.fastq.gz SRR10323947_1.trim.fastq.gz SRR10323947_1un.trim.fastq.gz SRR10323947_2.trim.fastq.gz SRR10323947_2un.trim.fastq.gz SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15

#Here are the arguements: 
#TrimmomaticPE: Started with arguments:
# SRR10323947_1.fastq.gz SRR10323947_2.fastq.gz SRR10323947_1.trim.fastq.gz SRR10323947_1un.trim.fastq.gz SRR10323947_2.trim.fastq.gz #SRR10323947_2un.trim.fastq.gz SLIDINGWINDOW:4:20 MINLEN:25 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15
#Using PrefixPair: 'AGATGTGTATAAGAGACAG' and 'AGATGTGTATAAGAGACAG'
#Using Long Clipping Sequence: 'GTCTCGTGGGCTCGGAGATGTGTATAAGAGACAG'
#Using Long Clipping Sequence: 'TCGTCGGCAGCGTCAGATGTGTATAAGAGACAG'
#Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTCCGAGCCCACGAGAC'
#Using Long Clipping Sequence: 'CTGTCTCTTATACACATCTGACGCTGCCGACGA'
#ILLUMINACLIP: Using 1 prefix pairs, 4 forward/reverse sequences, 0 forward only sequences, 0 reverse only sequences
#Quality encoding detected as phred33
#Input Read Pairs: 1086157 Both Surviving: 670281 (61.71%) Forward Only Surviving: 393675 (36.24%) Reverse Only Surviving: 13161 (1.21%) #Dropped: 9040 (0.83%)
#TrimmomaticPE: Completed successfully

```


You can see the tha nextera adapter content is now removed. There are still quality control concerns, but since the authors did not make any adjusttments, neither will we. 
![After cleaning up the adapter content](cleaned_adapters.png)

The next step is to download breseq:
```{r}
#conda -c bioconda install breseq

#Actually running it for a single sample:
# breseq -p -r sequence.gb -o /home/elyall_bmeg22/final_project/paired/trimmed_files/breseq_run/ SRR10323947_1.trim.fastq.gz SRR10323947_2.trim.fastq.gz
```

Making a textfile containing all of the SRA names
```{r}
#First we have to download the acession list as a txt file form NCBI
#https://www.ncbi.nlm.nih.gov/sra?term=SRP226618

#creating a .txt file containing all of the SRA #'s for each sample: 
nano SRA_seqs.txt #We then copy paste all of the SRA #'s names into this file 

```

Now, we are going to generate a pipline that downloads the fastQ files, runs fastQC on each of these, and then runs trimmomatic on each sample. 
Normaly we would use a job scheduler to do this, but apparently our server can't do this.
So we will used Dr. De Boer's script which runs a script for each line in an input file
The scripe is here: (https://raw.githubusercontent.com/BMEGGenoInfo/Assignments/main/Assignment_2/runTheseJobsSerially.sh

```{r}
#Downloading Carl's script: 
wget https://raw.githubusercontent.com/BMEGGenoInfo/Assignments/main/Assignment_2/runTheseJobsSerially.sh
#Getting permissions to run it:
chmod +x runTheseJobsSerially.sh
#Creating a pipeline called down_trim_pipeline.sh that downloads files from NCBI, runs fastQC and then runs trimmomatic on them. 

#First we make the name of the pipeline: 
nano down_trim_pipeline.sh

#Then we get permission on it
chmod +x down_trim_pipeline.sh
#Then we rune it!
./runTheseJobsSerially.sh ./pipelines/down_trim_pipeline.sh SRA_seqs.txt

#Then, we start adding to it!


#!/bin/bash
set -e # this makes the whole script exit on any error.
sample=$1
mkdir -p fastq_samples # making a directory to add the fastq files
mkdir -p fastqc_files #making a directory for fastqc_files
mkdir -p downtrim_logfiles
#Now, going to run the pipeline: 

echo running pipeline for $sample
if [ ! -e downtrim_logfiles/$sample.fastq.done ] 
then
        echo Downloading $sample
        #Command download sample:
        fastq-dump $sample --split-files --outdir fastq_samples/
        touch downtrim_logfiles/$sample.fastq.done #add a flag to say fastq downloaded.
else
        echo Already performed fastqc of $sample
        fi
if [ -e downtrim_logfiles/$sample.fastq.done ] #if the fastq is done,do fastqc
then
        echo Running fastqc with $sample
        mkdir fastqc_files/$sample
        fastqc $fastq_samples/{sample}_1.fastq --outdir fastqc_files/$sample/
        fastqc $fastq_samples/{sample}_2.fastq --outdir fastqc_files/$sample/
        touch downtrim_logfiles/$sample.fastqc.done
else
        echo: FastQC was not made sucessfully.
        fi

```

Next, we can make sure all the fastq files are gzipped before running trimmomatic:
```{r}
#gzip fastq_samples/*.fastq

#Making the name of the trimmomatic pipeline:
nano pipelines/trimmomatic.sh
chmod +x trimmomatic.sh

#running the pipeline: 

        

```

Now making a pipeline to trim all the samples: 
```{r}
#!/bin/bash
set -e # this makes the whole script exit on any error.
sample=$1
mkdir -p trimmomatic_files
if [ ! -e downtrim_logfiles/$sample.trim.done ]
then
        mkdir trimmomatic_files/$sample
        #trimmomatic PE $fastq_samples/${sample}_1.fastq.gz fastq_samples/${sample}_2.fastq.gz trimmomatic_files/$sample/${sample}_1.trim.fastq.gz trimmomatic_files/$sample/${sample}_1un.trim.fastq.gz trimmomatic_files/$sample/${sample}_2.trim.fastq.gz trimmomatic_files/$sample/${sample}_2un.trim.fastq.gz SLIDINGWINDOW:4:15 MINLEN:25 LEADING:3 TRAILING:3 ILLUMINACLIP:NexteraPE-PE.fa:2:40:15

        touch downtrim_logfiles/$sample.trimmed.done
else
        echo:trimmomatic failed
        fi
```

SRR10323947_1.fastq.gz
SRR10323947_1.fastq.gz


```{r}
#Now we make a pipeline to run breseq:
nano pipelines/breseq_pipe.sh
chmox +x breseq_pipe.sh
#The pipeline starts below: 

#!/bin/bash
set -e # this makes the whole script exit on any error.
sample=$1
mkdir -p breseq_files
if [ -e downtrim_logfiles/$sample.trimmmed.done ]
then
        #Going to run breseq on it
        mkdir -p breseq_files/$sample
        #Running the breseq command:
        # breseq -p -r sequence.gb -o /home/elyall_bmeg22/final_project/breseq_files/$sample/ trimmomatic_files/$sample/${sample}_1.trim.fastq.gz trimmomatic_files/$sample/${sample}_2.trim.fastq.gz
        touch downtrim_logfiles/$sample.breseq.done
else
        echo:breseq failed
        fin

    
```

